File: prak3_python/Pc.py
from Processor import Processor
from Disk import Disk
from Ram import Ram

class Pc:

    def __init__(self, processor, disk, ram):
    #konstruktor
        self.processor = processor
        self.disk = disk
        self.ram = ram
        self.totalPrice = int(processor.getPrice()) + int(disk.getPrice()) + int(ram.getPrice())

    #set
    def setProcessor(self, processor):
        self.processor = processor
    
    def setDisk(self, disk):
        self.disk = disk

    def setRam(self, ram):
        self.ram = ram

    def updateTotalPrice(self):
        self.totalPrice = int(self.processor.getPrice()) + int(self.disk.getPrice()) + int(self.ram.getPrice())

    #get
    def getProcessor(self):
        return self.processor

    def getDisk(self):
        return self.disk
    
    def getRam(self):
        return self.ram
    
    def getTotalPrice(self):
        return self.totalPrice

    #method
    def show(self):
        print("Processor")
        print("- Name : " + self.processor.getName())
        print("- Price : " + str(self.processor.getPrice()))
        print("Disk")
        print("- Type : " + self.disk.getType())
        print("- Capacity : " + self.disk.getCapacity())
        print("- Price : " + str(self.disk.getPrice()))
        print("Ram")
        print("- Capacity : " + self.ram.getCapacity())
        print("- Price : " + str(self.ram.getPrice()))
        print("totalPrice : " + str(self.getTotalPrice()))

File: prak3_php/Pc.php
<?php

include "Processor.php";
include "Disk.php";
include "Ram.php";

class Pc{
    
    private $processor;
    private $disk;
    private $ram;
    private $totalPrice;

    //konstruktor
    function __construct($processor, $disk, $ram){
        $this->processor = $processor;
        $this->disk = $disk;
        $this->ram = $ram;
        $this->totalPrice = $processor->getPrice() + $disk->getPrice() + $ram->getPrice();
    }


    //set
    public function setProcessor($processor){
        $this->processor = $processor;
    }

    public function setDisk($disk){
        $this->disk = $disk;
    }

    public function setRam($ram){
        $this->ram = $ram;
    }

    public function updateTotalPrice(){
        $this->totalPrice = $this->processor->getPrice() + $this->disk->getPrice() + $this->ram->getPrice();
    }

    //get
    public function getProcessor(){
        return $this->processor;
    }

    public function getDisk(){
        return $this->disk;
    }

    public function getRam(){
        return $this->ram;
    }

    public function getTotalPrice(){
        return $this->totalPrice;
    }

    //methods
    public function show(){
        echo "Processor<br/>";
        echo "- Name : ".$this->processor->getName()."<br/>";
        echo "- Price : ".$this->processor->getPrice()."<br/>";
        echo "Disk<br/>";
        echo "- Type : ".$this->disk->getType()."<br/>";
        echo "- Capacity : ".$this->disk->getCapacity()."<br/>";
        echo "- Price : ".$this->disk->getPrice()."<br/>";
        echo "Ram<br/>";
        echo "- Capacity : ".$this->ram->getCapacity()."<br/>";
        echo "- Price : ".$this->ram->getPrice()."<br/>";
        echo "totalPrice : ".$this->getTotalPrice()."<br/>";
    }


    //destruktor
    function __destruct(){
    }

}
    

?>